[14.0][MIG] account_bank_statement_import

alexis-via opened the pull request.

alexis-via committed:
Import account_bank_statement_import by Odoo SA

Import code just before commit 9ba8734f15e1a292ca27b1a026e8366a91b2a8c9
that moved the module account_bank_statement_import to Odoo Enterprise
Only modifications :
- add author (Odoo SA) and license (LGPL) keys in manifest
- add copyright headers Odoo SA all main files

alexis-via committed:
[MIG] account_bank_statement_import to v14

Module renamed to account_statement_import to avoid conflit with Odoo enterprise
Add support for multi-account statement files
Integrate the feature provided by the module account_bank_statement_import_save_file
Improve error messages
Remove dead or annoying features
Improve code !

alexis-via committed:
Black/isort

alexis-via committed:
Improve an error message

njeudy reviewed (approved):
test and ok for me thanks @alexis-via

i-vyshnevska commented on account_statement_import/i18n/account_bank_statement_import.pot:
```diff
@@ -0,0 +1,800 @@
+# Translation of Odoo Server.
```
remove file

i-vyshnevska commented on account_statement_import/wizard/account_statement_import.py:
```diff
@@ -0,0 +1,370 @@
+                    - 'unique_import_id': string
+                    -o 'account_number': string
+                        Will be used to find/create the res.partner.bank in odoo
+                    -o 'note': string
```
field "note" was removed from the model

i-vyshnevska reviewed (changes requested):
@alexis-via you need to keep the initial commit history

luc-demeyer commented on account_statement_import/wizard/account_statement_import.py:
```diff
@@ -0,0 +1,370 @@
+            # dispatch to reconciliation interface
+            line_ids = self.env["account.bank.statement.line"].search(
+                [("statement_id", "in", result["statement_ids"])]
+            )
```
search returns recordset -> change to search([...]).ids

alexis-via committed:
[FIX] line_ids was storing recordset

Thanks to Luc de Meyer for pointing the bug.

alexis-via committed:
Update comment: note field was removed from account.bank.statement.line

Remove POT file with bad filename
Thanks to Iryna Vyshnevska for pointing this out.

luc-demeyer commented:
@alexis-via 
note field has not been removed but is renamed to 'narration'

pedrobaeza commented:
Some git work is needed before merging:

- Change first commit author to be something like info@odoo.com for changing attribution.
- Pre-commit stuff should be before doing any change.
- Squash unneeded changes.

@alexis-via do you want me to do these changes?

alexis-via commented:
@pedrobaeza Yes, go ahead !

fclementic2c commented:
Hello, I would appreciate to have this PR merged quickly if possible. Thanks @alexis-via

pedrobaeza commented:
As I can't push to this branch being a corporate branch (don't know why GitHub applies such constraint), I have opened #367, and I will directly merge it as this one has enough approvals.
